api/envoy/config/listener/v3/listener.proto

```diff
@@ -93,6 +95,23 @@ message Listener {
   message InternalListenerConfig {
   }
 
+  // Filter chain with a name.
+  message NamedFilterChain {
+    // Filter chain name.
+    string name = 1 [(validate.rules).string = {min_len: 1}];
```

Comment:
Can you make it clear that this name must be unique and what happens if it's not?

Reply:
I think we have to accept duplicates because of backwards compatibility, but if the matcher is set, we can start rejecting duplicates.